Modify cfn-lint custom script to restrict checks against valid cloudformation extensions

prashant0085 opened the pull request:
<!-- Please update the GitHub commit summary with a 1-line <80 char desc -->

### Modify cfn-lint custom script to restrict checks against valid cloudformation extensions

Task Link: <!-- Provide link to Trello card/GitHub Issue/etc --> N/A

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Developer Checklist
- [ ] My change requires a change to the documentation.
- [x] I have updated my branch with the latest changes in master.
- [x] I have communicated with the team any changes that might affect work in progress. 
- [x] I have updated Trello to reflect the current state of the feature. 

## Documentation Checklist
- [ ] I have updated the necessary README's
- [ ] I have updated the Runbook

## Description of Changes
- Modify cfn-lint custom script to check only yaml, yml and json

## Context
- False positives triggered by cfn-lint PR check. A python file containing the CF template marker(`AWSTemplateFormatVersion`) used in this script was flagged  and ran against cfn-lint.
![image](https://user-images.githubusercontent.com/3526167/94039973-00720800-fde6-11ea-948a-bc8b196db560.png)


## Description of Testing Process
- Verified locally

<!-- How the changes were tested ( this can be a unit test or a functional test depending on the changes ) -->

Commit:
added files for which cfn-lint should run on

Commit:
fixed the path to cfn templates

hanumantharaomvl commented:
@prashant0085 we are already handling the type of files for cfn-lint here. https://github.com/Flux7Labs/central-repo-git-hooks/blob/master/.pre-commit-hooks.yaml

vishnu-narayanan committed:
Revert "fixed the path to cfn templates"

This reverts commit b61ce3f462e08d42463a0bde4349edc58025f5b5.

vishnu-narayanan committed:
Modify cfn-lint custom script to check only yaml, yml and json

context: False positives triggered by cfn-lint PR check. A python
file containing the CF template marker used in this script was
flagged and ran against cfn-lint.

vishnu-narayanan committed:
Revert "added files for which cfn-lint should run on"

This reverts commit 4dcffbc0ca8d25da04608e38f6c07e0cf9a1dfa5.

vishnu-narayanan commented:
@hanumantharaomvl The custom script used as entry point for the `cfn-lint` check was flagging a python file since it does not handle extensions.

sirhc commented on hooks/scripts/cfn_lint.sh:
```diff
@@ -11,7 +11,7 @@ set -e
 CF_CHECK='AWSTemplateFormatVersion'
 current_commit=$(git rev-parse HEAD)
-templates=$(find `pwd` -type f -exec grep -l $CF_CHECK {} \;)
+templates=$(find `pwd` -type f -exec grep -l $CF_CHECK --include \*.yaml --include \*.yml --include \*.json {} \;)
```
Personally, I'd move the file name filtering to the `find` command instead of the `grep` command, but I'm not sure it makes much of a difference.

Is `CF_CHECK` used anywhere else? There doesn't seem much use in creating a variable that will be used only once on a line almost immediately following.

vishnu-narayanan replied:
@sirhc Thank you for pointing it out. It would have more or less the same effect here I guess.

`CF_CHECK` is used only here. Keeping it since I plan to raise a separate discussion/PR around overhauling `pre-commit`scripts.

vishnu-narayanan commented:
Thank you for the review @sirhc @hanumantharaomvl
